### ISSUE TITLE
State constraints

### ISSUE THREAD

**Issue by aaronwolen** [home-assistant/appdaemon#162]
Would it be possible to generalize `constrain_input_select` and `constrain_input_boolean` to `constrain_state`?

I've run into a number of scenarios in which this would be useful. As a simple example, I only want the motion lights app to run if the specified light is off.

**Comment by acockburn** [comment]
I'm thinking about adding custom constraints which would allow you to implement something similar yourself - something like:

```python
def my_constraint():
  if self.get_state("light.living_room") == "on":
    return True
  else:
  return False
```

Then you would add `my_constraint()` as a constraint to the callback. When this is in place, all the existing constraints will be re-implemented as fixed custom constraints, so at that point, it would also be easy to add a `constrain_state()`

**Comment by aaronwolen** [comment]
Wow, sounds awesome! 👍

**Comment by acockburn** [comment]
Custom constraints are implemented in 3.0 so closing.

**Comment by freekode** [comment]
Where I can find documentation how to create custom constraints?
